subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/InHouseInstrumentationTestRunner.kt

```diff
@@ -165,14 +169,12 @@ abstract class InHouseInstrumentationTestRunner :
     private val reportHttpClient: OkHttpClient by lazy { createReportHttpClient() }
 
     protected abstract val metadataToBundleInjector: TestMetadataInjector
+    private val testMetadataValidators: MutableList<TestMetadataValidator> = mutableListOf()
 
     abstract fun createRunnerEnvironment(arguments: Bundle): TestRunEnvironment
 
-    private fun injectTestMetadata(arguments: Bundle) {
-        val isRealRun = !arguments.containsKey(FAKE_ORCHESTRATOR_RUN_ARGUMENT)
-        if (isRealRun) {
-            metadataToBundleInjector.inject(arguments)
-        }
+    protected fun addTestValidator(validator: TestMetadataValidator) {
```

Comment:
`protected` could be invoked at any time - it may lead to incorrect usages of that function.
I prefer to have a function in which you will return your additional test validators

Reply:
Fixed. Now a user must override the validator.